linode/instancedisk/framework_resource.go

```diff
@@ -0,0 +1,473 @@
+		plan.StackScriptID.ValueInt64(), &resp.Diagnostics,
+	)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createOpts := linodego.InstanceDiskCreateOptions{
+		Filesystem:    plan.Filesystem.ValueString(),
+		Image:         plan.Image.ValueString(),
+		Label:         plan.Label.ValueString(),
+		Size:          diskSize,
+		StackscriptID: stackScriptID,
+	}
+
+	resp.Diagnostics.Append(plan.AuthorizedKeys.ElementsAs(ctx, &createOpts.AuthorizedKeys, false)...)
+	resp.Diagnostics.Append(plan.AuthorizedUsers.ElementsAs(ctx, &createOpts.AuthorizedUsers, false)...)
+	resp.Diagnostics.Append(plan.StackScriptData.ElementsAs(ctx, &createOpts.StackscriptData, false)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if plan.RootPass.IsNull() {
+		createOpts.RootPass = helper.FrameworkCreateRandomRootPassword(&resp.Diagnostics)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	} else {
+		createOpts.RootPass = plan.RootPass.ValueString()
+	}
+
+	p, err := client.NewEventPoller(ctx, linodeID, linodego.EntityLinode, linodego.ActionDiskCreate)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to Poll for Events", err.Error())
+		return
+	}
+
+	tflog.Debug(ctx, "client.CreateInstanceDisk(...)")
+	disk, err := client.CreateInstanceDisk(ctx, linodeID, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to Create Disk on Linode Instance %d", linodeID),
+			err.Error(),
+		)
+		return
+	}
+
+	// Add resource to TF states earlier to prevent
+	// dangling resources (resources created but not managed by TF)
+	AddDiskResource(ctx, *disk, resp, plan)
+
+	ctx = tflog.SetField(ctx, "disk_id", disk.ID)
+
+	event, err := p.WaitForFinished(ctx, timeoutSeconds)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to Wait for the Instance Shutdown Event (%d)", event.ID),
+			err.Error(),
+		)
+	}
+
+	if _, err := client.WaitForInstanceDiskStatus(
+		ctx, linodeID, disk.ID, linodego.DiskReady, timeoutSeconds,
+	); err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to Wait for Disk (%d) to be Ready", disk.ID), err.Error(),
+		)
+	}
+
+	// get latest status of the disk
+	tflog.Trace(ctx, "client.GetInstanceDisk(...)")
+	disk, err = client.GetInstanceDisk(ctx, linodeID, disk.ID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to Get Disk %d of Linode Instance %d", disk.ID, linodeID),
+			err.Error(),
+		)
+	}
+
+	// IDs should always be overridden during creation (see #1085)
+	// TODO: Remove when Crossplane empty string ID issue is resolved
+	plan.ID = types.StringValue(strconv.Itoa(disk.ID))
+
+	plan.FlattenDisk(disk, true)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *Resource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	tflog.Debug(ctx, "Read "+r.Config.Name)
+
+	var state ResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
```

Comment:
Could we call `helper.FrameworkAttemptRemoveResourceForEmptyID(...)` here for the Crossplane workaround?